src/Data/List/Fresh/NonEmpty.agda

```diff
@@ -0,0 +1,52 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- A non-empty fresh list
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.List.Fresh.NonEmpty where
+
+open import Level using (Level; _⊔_)
+open import Data.List.Fresh as List# using (List#; []; cons; fresh)
+open import Data.Maybe.Base using (Maybe; nothing; just)
+open import Data.Nat using (ℕ; suc)
+open import Data.Product using (_×_; _,_)
+open import Data.Unit.Polymorphic.Base using (⊤; tt)
+open import Relation.Binary as B using (Rel)
+
+private
+  variable
+    a r : Level
+    A : Set a
+
+module _ {a} (A : Set a) (R : Rel A r) where
+
+  infixr 5 _∷#⁺_
+  record List#⁺ : Set (a ⊔ r) where
+    constructor _∷#⁺_
+    field
+      head : A
+      tail : List# A R
+      {rel} : fresh A R head tail
+
+  open List#⁺
+
+  uncons : List#⁺ → A × List# A R
```

Comment:
I don't think you mean to put these under `module _ {a} (A : Set a) (R : Rel A r) where`. Currently it means that every time you use operations like `uncons` you'll have to pass in the `A` and `R` explicitly, and the singleton list syntax `[_]` will be unusable. Better to make `R` a variable and instead write:
```
uncons : List#⁺ A R → A × List# A R
```
outside the anonymous module. This is more readable as well.